src/main/java/cam72cam/immersiverailroading/entity/EntityRollingStock.java

```diff
@@ -147,7 +149,7 @@ public boolean attackEntityFrom(DamageSource damagesource, float amount) {
 		}
 		
 		if (damagesource.isExplosion()) {
-			if (amount > 5) {
+			if (amount > 5 && !(Config.damage.trainExplosionDamage && damagesource.getTrueSource() instanceof EntityMob)) {
```

Comment:
This logic is wrong

```
! ( config_true  && isMob_true  ) = false
! ( config_true  && is_Mob_false) = true
! ( config_false && isMob_true  ) = true
! ( config_false && is_Mob_false) = true
```

Fixed:

(amount > 5 && (Config.damage.trainExplosionDamage || !damagesource.getTrueSource() instanceof EntityMob))

```
( config_true  || !isMob_true  ) = true
( config_true  || !is_Mob_false) = true
( config_false || !isMob_true  ) = false
( config_false || !is_Mob_false) = true
```